### ISSUE TITLE
Window List Preferences: Sort by Recently Used, Rearrange when windows grouped

### ISSUE THREAD

**Issue by mate-user** [mate-desktop/mate-desktop#394]
#### Expected behaviour

Window Grouping currently sorts windows alphabetically, but this is not shown.
It would be nice if there were an option to show the list in the order I had the windows open, and to reverse it too. So for example I have 17 Firefox windows open now and when I click the grouped window button, I'd like to see my main Firefox window which is on the left side usually when ungrouped to show as the first entry starting from the bottom.
Windows on the left at the bottom, windows on the right at the top, and make this reversible.
I also want to be able to rearrange the order of the windows in grouped mode so it's reflected when I ungroup windows.
When grouped, I would expect it to show in the same order as when it is ungrouped.

![image](https://user-images.githubusercontent.com/41811862/59372967-efb09680-8d48-11e9-933f-90679d481f66.png)

![image](https://user-images.githubusercontent.com/41811862/59372721-66995f80-8d48-11e9-87bf-8015704fb9d0.png)

#### Actual behaviour

The grouped list shows alphabetically.

#### Steps to reproduce the behaviour

Open lots of Firefox windows with different tab names, then set to group windows when space is limited or always group.

#### MATE general version

1.22.1

#### Linux Distribution

Solus 4

**Comment by vkareh** [comment]
Relevant:
* https://gitlab.gnome.org/GNOME/libwnck/issues/6
* https://gitlab.gnome.org/GNOME/libwnck/issues/44

**Comment by raveit65** [comment]
> So for example I have 17 Firefox windows open now....

Amazing, this eats a lot of memory...
Why not using 1 Firefox window with 17 tabs :)

**Comment by mate-user** [comment]
I keep separate windows for different tasks/things like YouTube videos, news etc to check later.
If I had 1 Firefox window, it would have 3,500 tabs in total haha :P 
My main Firefox window has 3,100 tabs built up over the last 1-2 years!

**Comment by raveit65** [comment]
Who use  3500 tabs?

**Comment by mate-user** [comment]
News stories and blog posts I opened to read later. I think I have a tab procrastination problem haha!

**Comment by Xmetalfanx** [comment]
I'd consider really organizing Bookmarks ... via a https://youtu.be/bGTBH9yr8uw?t=987 and TuxDigital (he may be doing a separate video on how to do this too) you can use stuff like keywords in the bookmarks to say open Reddit when "R" + Enter is pressed (vs searching for R in your default search engine) ... that just one example

sorry Devs, off topic i know

**Comment by flexiondotorg** [comment]
3500 tabs in 17 browser windows collected over 2 years? I'm not sure there is a technical solution here, should we start an intervention? 😁

**Comment by mate-user** [comment]
Yeah I've bookmarked 100's of tabs before and mostly forgotten about them, I'll do that if this gets too distracting.
